mlrun/datastore/datastore_profile.py

```diff
@@ -295,6 +295,30 @@ def secrets(self) -> dict:
         return res if res else None
 
 
+class DatastoreProfileHdfs(DatastoreProfile):
+    type: str = pydantic.Field("hdfs")
+    _private_attributes = "token"
+    host: typing.Optional[str] = None
+    port: typing.Optional[int] = None
+    http_port: typing.Optional[int] = None
+    user: typing.Optional[str] = None
+
+    def secrets(self) -> dict:
+        res = {}
+        if self.host:
+            res["HDFS_HOST"] = self.host
+        if self.port:
+            res["HDFS_PORT"] = self.port
+        if self.port:
+            res["HDFS_HTTP_PORT"] = self.http_port
+        if self.user:
+            res["HDFS_USER"] = self.user
+        return res or None
+
+    def url(self, subpath):
```

Comment:
You ether provide port and host as part of the url , or as part of the storage parameters in fsspec. I think url should return "webhdfs://' or 'webhdfs:/'